python/ray/actor.py

```diff
@@ -254,9 +372,105 @@ def __ray_restore_from_checkpoint__(cls, pickled_checkpoint):
                 # TODO(rkn): It's possible that this will cause problems. When
                 # you unpickle the same object twice, the two objects will not
                 # have the same class.
-                actor_object = pickle.loads(checkpoint)
+                actor_object = checkpoint
             return actor_object
 
+        def __ray_checkpoint__(self, task_counter, previous_object_id):
+            """Save or resume a stored checkpoint.
+
+            This task checkpoints the current state of the actor. If the actor
+            has not yet executed to `task_counter`, then the task instead
+            attempts to resume from a saved checkpoint that matches
+            `task_counter`. If the most recently saved checkpoint is earlier
+            than `task_counter`, the task requests reconstruction of the tasks
+            that executed since the previous checkpoint and before
+            `task_counter`.
+
+            Args:
+                self: An instance of the actor class.
+                task_counter: The index assigned to this checkpoint method.
+                previous_object_id: The dummy object returned by the task that
+                    immediately precedes this checkpoint.
+
+            Returns:
+                A bool representing whether the checkpoint was successfully
+                    loaded (whether the actor can safely execute the next task)
+                    and an Exception instance, if one was thrown.
+            """
+            previous_object_id = previous_object_id[0]
+            # Make sure that a previous object was given.
+            if previous_object_id is None:
+                return False
```

Comment:
If this code path is ever taken, that's a bug, right? Or can it happen normally?

Reply:
Oh yeah, that's right.